SalesforceCommerceCloud/pwa-kit
# Merge `feature/shopper-context` into `v5/template-retail-react-app` branch

# Description



Adds a Shopper Context integration to the Retail React App. OOTB shopper context can be set via URL query params. This also adds a Page Designer integration to the PLP page to display a promo banner where in the demo site it will display an instagram promo banner when `sourceCode=instagram`.

This PR is a culmination of PRs previously approved:
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1986
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2016
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2024
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2064
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2086

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1986
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2016
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2024
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2064
- https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2086

# How to Test-Drive This PR

This branches code has been deployed to https://runtime-admin-jinsu.mobify-storefront.com/salesforce-internal/cloud-jinsu/env2

- Navigate to the PLP page: https://cloud-jinsu-env2.mobify-storefront-staging.com/global/en-GB/category/womens-clothing-tops
- open the developer console and switch to the *Network* tab
- Refresh the page
  - Verify the `/shopper-context` API call returns a 404
  - Verify no `/shopper-experience` API calls are made
- Set the shopper context to `?sourceCode=fb`: https://cloud-jinsu-env2.mobify-storefront-staging.com/global/en-GB/category/womens-clothing-tops?sourceCode=fb
  - Verify a `PUT /shopper-context` API call was made and returns 201
  - Verify a `GET /shopper-context` API call was made and returns a 200 and the response body is: `{"sourceCode":"fb"}`
  - Verify 2 `/shopper-experience` API calls were made for `instagram-promo-banner-desktop` and `pages/instagram-promo-banner-mobile` and both return a 500 (we should expect 404 but a bug ticket has been created to resolve this [see slack thread for more details](https://salesforce-internal.slack.com/archives/C01RW0121HR/p1727793546564629))
- Set the shopper context to `?sourceCode=instagram`: https://cloud-jinsu-env2.mobify-storefront-staging.com/global/en-GB/category/womens-clothing-tops?sourceCode=instagram
  - Verify a `PATCH /shopper-context` API call was made and returns 200
  - Verify a `GET /shopper-context` API call was made and returns a 200 and the response body is: `{"sourceCode":"instagram"}`
  - Verify 2 `/shopper-experience` API calls were made for `instagram-promo-banner-desktop` and `pages/instagram-promo-banner-mobile` and both return a 200
  - Verify a promo banner is displayed at the top of the page
  - Verify the pricing for all products have changed and each of the products have a new promo callout message "Exclusive for Instagram users"

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
